File: data/src/main/java/com/icehousecorp/maunorafiq/data/weather/repository/datasource/WeatherDataStoreFactory.java
package com.icehousecorp.maunorafiq.data.weather.repository.datasource;

import javax.inject.Inject;
import javax.inject.Singleton;

import retrofit2.Retrofit;

/**
 * Created by Raffi on 11/25/2016.
 */
@Singleton
public class WeatherDataStoreFactory {

    private Retrofit retrofit;

    @Inject
    public WeatherDataStoreFactory(Retrofit retrofit) {
        this.retrofit = retrofit;
    }

    public WeatherDataStore create () {
        WeatherDataStore weatherDataStore;

        weatherDataStore = new CloudWeatherDataStore(retrofit);

        return weatherDataStore;
    }
}

File: data/src/main/java/com/icehousecorp/maunorafiq/data/weather/repository/datasource/WeatherDataStoreFactory.kt
package com.icehousecorp.maunorafiq.data.weather.repository.datasource

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Raffi on 11/25/2016.
 */
@Singleton
class WeatherDataStoreFactory
@Inject
constructor() {

    fun create(): WeatherDataStore {
        val weatherDataStore: WeatherDataStore

        weatherDataStore = CloudWeatherDataStore()

        return weatherDataStore
    }
}
